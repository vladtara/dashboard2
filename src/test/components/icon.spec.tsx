import { fireEvent, render, screen } from "@testing-library/react";
import Icon, { IconButton } from "../../components/icon";

const props = {
  name: "bug_report",
  size: "20px",
  onClick: jest.fn(),
};

it("Icon test", () => {
  const { baseElement } = render(<Icon name={props.name} size={props.size} />);
  expect(baseElement).toMatchSnapshot();
});

it("Icon test (no size)", () => {
  const { baseElement } = render(<Icon name={props.name} />);
  expect(baseElement).toMatchSnapshot();
});

it("IconButton test", () => {
  const iconButton = render(
    <IconButton icon={props.name} onClick={props.onClick} />,
  );

  expect(iconButton.baseElement).toMatchSnapshot();

  fireEvent.click(screen.getByText(/bug_report/i));
  expect(props.onClick).toHaveBeenCalledTimes(1);
});
