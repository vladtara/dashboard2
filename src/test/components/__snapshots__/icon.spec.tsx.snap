// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon test (no size) 1`] = `
<body>
  <div>
    <i
      class="sc-gsnTZi ctUcHQ"
    >
      bug_report
    </i>
  </div>
</body>
`;

exports[`Icon test 1`] = `
<body>
  <div>
    <i
      about="20px"
      class="sc-gsnTZi iGhKXa"
    >
      bug_report
    </i>
  </div>
</body>
`;

exports[`IconButton test 1`] = `
<body>
  <div>
    <button
      class="sc-bczRLJ liKoho"
    >
      <i
        class="sc-gsnTZi ctUcHQ"
      >
        bug_report
      </i>
    </button>
  </div>
</body>
`;
